### PR TITLE
adds optional and experimental llm provider Mistral API

### PR DESCRIPTION
**Pull request by xbasset**

### CODE REVIEW

File: docs/4._LLM_Providers/openai.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Setup the API key in your `.env` file:
+
+```bash
+OPENAI_API_KEY=your-api-key
+```
+
+!!! note "Switching from OpenAI to Azure OpenAI Services"
+    This is made easy by changing the `MAIN_MODEL_API_TYPE` in your `.env` file.
+
+
```
Comment (KellyRousselHoomano) [inline docs/4._LLM_Providers/openai.md:12]: Maybe we should also mention "backup engine" here.

File: docs/4._LLM_Providers/mistral.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+In your `.env` file, add the following:
+
+```bash
+MISTRAL_API_KEY=your-api-key
+```
+
+## Coverage
+
+> For the assessment, `mistral-medium`has been tested on the most challenging task: `"ANSWER_USER"`, which is the most challenging task.
```
Comment (KellyRousselHoomano) [inline docs/4._LLM_Providers/mistral.md:20]:
```suggestion
> For the assessment, `mistral-medium`has been tested on feature: `"ANSWER_USER"`, which is the most challenging feature.
```

File: docs/4._LLM_Providers/openai.md (hunk 1 above)
Comment (xbasset) [reply docs/4._LLM_Providers/openai.md:22]: done

File: backend/app/models/llm_calls/mojodex_mistral.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            responses=[]
+
+            try:                
+                # ensure if stream is true, n_responses = 1
+                if stream and n_responses != 1:
+                    n_responses = 1
+                    logging.warning("n_responses must be 1 if stream is True")
+
+                
+                client = MistralClient(api_key=self.api_key)
```
[the remark this answers is not in the document]
Comment (xbasset) [reply backend/app/models/llm_calls/mojodex_mistral.py]: done

**Review by KellyRousselHoomano — APPROVED** [review]

**Pull request merged**